Doxyfile.in

```diff
@@ -780,9 +780,11 @@ WARN_LOGFILE           =
 # spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  = @CMAKE_CURRENT_SOURCE_DIR@/libzeth/ \
+INPUT                  = @CMAKE_CURRENT_SOURCE_DIR@/libtool/ \
```

Comment:
I'm curious as to why we need to explicitly list all of the subdirectories here.  Have you tried giving the repo root here, and adding `depends` (and any other directories) to the `EXCLUDE` variable?

Reply:
>  Have you tried giving the repo root here, and adding depends (and any other directories) to the EXCLUDE variable?

Yes. Happy to switch to this, I don't have strong preferences, tbh.
The reason I listed files in INPUT explicitly instead is because our code is not very neatly packaged so far. Ideally, we'd have a `src/` folder, a `doc` folder etc etc,and we'd set the INPUT of the Doxyfile to `INPUT = src` and have a couple exclusion within this folder e.g. `EXCLUDE = src/tests` etc.
However, if we use the root dir as INPUT here, we need to exclude quite a few things: `depends`, `zeth_contracts`, `testdata`, all the `env` folders that may be in one's local repo etc. Furthermore, if we run tests locally, we generate lots of folders that also need to be excluded if we run `make docs` on a local version of the repo. So to keep things simple I preferred to list the `INPUT` to the folders of interest.

Reply:
> Yes. Happy to switch to this, I don't have strong preferences, tbh.

It may not be possible, but if we can configure this to keep so it automatically picks up new components then I think it's preferable.  It keeps the maintenance requirements low.

> However, if we use the root dir as INPUT here, we need to exclude quite a few things: depends, zeth_contracts, testdata, all the env folders that may be in one's local repo etc. Furthermore, if we run tests locally, we generate lots of folders that also need to be excluded if we run make docs on a local version of the repo. So to keep things simple I preferred to list the INPUT to the folders of interest.

I'm not sure what kind of patterns are allowable, but if we can exclude directories such as `*/build/*`, `*/node_modles/*`, `*/env/*` then perhaps we can exclude most of what we need and again keep the maintenance cost of this file down.

As long as our test data does not contain source code (.h, .cpp, etc files), does it need excluding?

Reply:
Yes, that's better you're right. I used `EXCLUDE_PATTERNS` to filter things out